test-network-function/certification/suite.go

```diff
@@ -87,25 +87,32 @@ func waitForCertificationRequestToSuccess(certificationRequestFunc func() bool,
 	return isCertified
 }
 
-//nolint:dupl
 func testContainerCertificationStatus() {
 	// Query API for certification status of listed containers
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestContainerIsCertifiedIdentifier)
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		env := configpkg.GetTestEnvironment()
-		containersToQuery := env.Config.CertifiedContainerInfo
-
-		if len(containersToQuery) == 0 {
+		containersToQuery := make(map[configsections.CertifiedContainerRequestInfo]bool)
+		for _, c := range env.Config.CertifiedContainerInfo {
+			containersToQuery[c] = true
+		}
+		if env.Config.CheckDiscoveredContainerCertificationStatus {
+			const redhatContainerCatalogRegistry = "https://catalog.redhat.com/software/containers/"
+			for _, cut := range env.ContainersUnderTest {
+				if cut.ImageSource.Registry != redhatContainerCatalogRegistry {
+					containersToQuery[configsections.CertifiedContainerRequestInfo{Repository: cut.ImageSource.Repository, Name: cut.ImageSource.Name}] = true
+				}
+			}
+		}
+		if len(containersToQuery) == 0 && !env.Config.CheckDiscoveredContainerCertificationStatus {
```

Comment:
If I understood the logic here correctly, it should be || (or), right?

Reply:
no, we call skip only if there is no query to make and we didn't check (based on prefix/registry) discovered ones. i will add a comment to the prefix check. that is being considered success result too.

Reply:
This has been removed as we are not sure about the registry and we may want to do the query anyways for better verification that considers other things like version platform etc.